### ISSUE TITLE
WebUI changes not propagated back to the app

### ISSUE THREAD

**Issue by Catfriend1** [syncthing/syncthing-android#1078]
According to bug report https://github.com/syncthing/syncthing-android/pull/1051#issuecomment-386451058
AudriusButkevicius: "I've also noticed that if you change settings in the web ui, they are not propagated back to the app. I thought the app handles configSaved events."

Suggestion to fix it:
- EventProcessor should react to "ConfigSaved" events.

**Comment by Catfriend1** [comment]
@AudriusButkevicius I did some tests and could easily reproduce the issue of configSaved not being handled and the app getting stuck if you press the "Restart Syncthing" button on the web UI.
Possible ways to solve this:
a) If the config is saved but syncthing core needs a restart, let the android app automatically do this. This would be my preferred way as it is easy and straightforward to implement.
b) Popping up the "Syncthing app needs a restart" notification in android's notification bar - as it's already done during the settings dialog for a postponed restart of syncthing. But then, we need to do an ugly workaround to also allow the user to restart syncthing core binary from the web UI (parse log for the process exit reason and restart the binary automatically from "SyncthingRunnable").
I'd like to have some feedback before I start to code please.

**Comment by AudriusButkevicius** [comment]
Sure, but that has nothing todo with stale ui state

**Comment by AudriusButkevicius** [comment]
To be honest, we should just restart syncthing when it needs it. Yet that could be weird, if yousave one folder and the UI goes down...

**Comment by Nutomic** [comment]
I think it would be possible to restart only the app in that case, and leave Syncthing running. Then data in the app would be up to date, and the user can still restart Syncthing when they are done. So basically you would only restart the RestApi class, but not SyncthingRunnable. Not sure how much effort that would be though.

**Comment by Catfriend1** [comment]
I made some draft code changes...
a) ConfigSaved triggers a ACTION_RESTART intent (like changing app's preferences) > found this too much as it's also happening if we change a folders option the app's UI (beside the intended restart when we use the webUI to change folder settings)
b) ConfigSaved tries to reinstantiate restApi and EventProcessor... we are not sure if we miss a sendConfig command from the app itself or an event will be fired after a regular sendConfig and ACTION_RESTART intent... hmm... not sure.
c) I found it working perfectly when a ConfigSaved() event triggers mApi.onWebGUIavailable(), as this function re-reads all systemversion, config, folder, device and gui settings stuff. This leaves the syncthing core untouched up and running. > Made a PR with it. I can change e.g. folder settings in the app's UI and some seconds later we will do a config reread "in silent" - doesn't hurt. If we go through the web UI and change folder settings, config will also be reread, this leads even to a correct outcome if I press back quickly and go to the same folder edit dialog of the folder I edited on the WebUI ... the switch toggles itself immediately so app UI reflects correct updated config state.
